internal/provider/resources/install/aws/aws_staged.go

```diff
@@ -0,0 +1,226 @@
+			},
+			"components": schema.SetAttribute{
+				ElementType:         types.StringType,
+				Required:            true,
+				MarkdownDescription: `Components to install (any of "iam-write", "inventory")`,
+				Validators: []validator.Set{
+					setvalidator.ValueStringsAre(
+						stringvalidator.OneOf(Components...),
+					),
+					setvalidator.SizeAtLeast(1),
+				},
+			},
+		},
+	}
+}
+
+func (r *StagedAws) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	data := internal.Configure(&req, resp)
+	if data != nil {
+		r.data = data
+	}
+}
+
+func (r *StagedAws) itemPath(component string, id string) string {
+	return fmt.Sprintf("integrations/aws/config/%s/%s", component, id)
+}
+
+func (r *StagedAws) fromComponentJson(data *StagedAwsModel, json *StagedAwsComponentJson, component string) {
+	if json.Label != nil {
+		data.Label = types.StringValue(*json.Label)
+	}
+	data.Components = append(data.Components, component)
+}
+
+func (r *StagedAws) fromJson(data *StagedAwsModel, json *StagedAwsJson) {
+	data.Components = []string{}
+	data.Label = types.StringNull()
+	data.ServiceAccountId = types.StringNull()
+
+	base, okBase := json.Config.Base[data.Id]
+	if okBase {
+		data.ServiceAccountId = types.StringValue(base.ServiceAccountId)
+	}
+
+	iamWriteJson, okIamWrite := json.Config.IamWrite[data.Id]
+	if okIamWrite {
+		r.fromComponentJson(data, &iamWriteJson, IamWrite)
+	}
+
+	inventoryJson, okInventory := json.Config.Inventory[data.Id]
+	if okInventory {
+		r.fromComponentJson(data, &inventoryJson, Inventory)
+	}
+}
+
+func (r *StagedAws) readState(ctx context.Context, diags *diag.Diagnostics, data *StagedAwsModel, state *tfsdk.State) {
+	var config StagedAwsJson
+	httpErr := r.data.Get("integrations/aws/config", &config)
+	if httpErr != nil {
+		diags.AddError("Error communicationg with P0", fmt.Sprintf("Unable to read AWS configuration, got error:\n%s", httpErr))
+		return
+	}
+	r.fromJson(data, &config)
+
+	// Save updated data into Terraform state
+	diags.Append(state.Set(ctx, data)...)
+}
+
+func (r *StagedAws) put(ctx context.Context, diags *diag.Diagnostics, plan *tfsdk.Plan, state *tfsdk.State, operation string) {
+	var data StagedAwsModel
+	diags.Append(plan.Get(ctx, &data)...)
+	if diags.HasError() {
+		return
+	}
+
+	report := func(component string, err error) {
+		if err != nil {
+			diags.AddError(fmt.Sprintf("Could not %s %s component", operation, component), fmt.Sprintf("Error: %s", err))
+		}
+	}
+
+	for _, component := range Components {
+		path := r.itemPath(component, data.Id)
+		if slices.Contains(data.Components, component) {
+			// We need to read back the entire configuration anyway (to get the base config), so ignore post responses
+			throwaway_response := struct{}{}
+			var json StagedAwsComponentJson
+			err := r.data.Put(path, &throwaway_response, &json)
```

Comment:
```suggestion
			err := r.data.Put(path, &struct{}{}, &json)
```

Is the goal of using the throwaway_response for self-documentation purposes? I think it's also pretty clear we don't need the struct if we inline it.

Reply:
yeah, exactly, just self-documentation